Creation of state and separation from view

Cracktuar opened the pull request:
- Rambling Summary -
  So I spent the last two days pulling apart the state from the view - now we have a collection of items that is completely independent from the collection of drawn items and can be swapped out freely.  This should make networking this easier.  At the very least, it should provide everyone with a nice starting point to begin network building stuff.
  
  -Testing -
  I ran through about half a game with Isaac, and tested re-roll + various curses and it all seemed OK but you might want to try with some of the other features.
- Functionality -
  Functionally, to an end user, it should be identical.  This was all backend changes.

NOTE: This breaks anything that relied on the old model where floors were items.

EDIT: Should solve issue #40

Cracktuar commented:
@Zamiell Here's the pull request I mentioned in the "State" issue.

Gama11 commented on item_tracker.py:
```diff
@@ -29,7 +32,7 @@ def __init__(self, id, x, y, index, shown, floor):
 
 # TODO: keep track of all curses
-class Floor:
+class aFloor:
```
Why `aFloor`?

Cracktuar replied:
It was a temporary change to keep it from conflicting with the Floor class in floor.py - I didn't want to remove the old floor class yet because I was still testing the new stuff. I forgot to clean that up when I removed everything from item_tracker.py.

Hyphen-ated commented:
I tried merging this but there were too many conflicts with the stuff gama11 and inschato wrote for me to confidently resolve them

Cracktuar commented:
Ahh shoot, you're right - I forgot that you can get nine lives twice.  I'll fix that later today.  I'll also try to merge what I wrote with what they wrote.

And originally guppy tracking worked by keeping a separate list and looping over that every time an item was picked up.  I replaced that with just looping over "collected_items" and counting the number of guppy items (but that's obviously wrong).  

How about just changing it so the item tracker keeps a set of items? That guarantees our count is of unique items, and avoids any weird issues that might come up if space bar items get broken down the line?

Hyphen-ated commented:
I'm not sure what keeping an additional set would be useful for besides just guppy tracking. It's not right for general stat tracking, because if you get like, two small rocks they give double the effect. 
We could change the current list of guppy items to be a set of guppy items, but does it really even matter? I guess it's more micro-efficient to query for set membership than iterating over a list of 6 elements

Cracktuar commented:
Ah, I meant swapping the guppy list to a guppy set.  It lets us just use the size of it to determine if we're guppy, because we won't end up double counting dead cats (for example).

Cracktuar committed:
Merge branch 'branch-to-merge'

Cracktuar committed:
Changed guppy_count to guppy_set

This change will let us avoid double-counting things like "dead cat"
that a player can obtain twice.

Cracktuar committed:
Dummied out generate_run_summary for now

Currently the run_summary stuff depends on the non-state model.  I
dummied it out so it doesn't accidentally crash the program.

Zamiell commented:
Cracktuar, did you update this pull request to make it so that it removes the conflicts that Hyphen mentioned earlier?
